cmd/conduit/main.go

```diff
@@ -0,0 +1,112 @@
+package main
+
+import (
+	"fmt"
+	"os"
+
+	log "github.com/sirupsen/logrus"
+	"github.com/spf13/cobra"
+
+	"github.com/algorand/indexer/conduit"
+	"github.com/algorand/indexer/config"
+)
+
+import (
+	_ "github.com/algorand/indexer/exporters/all"
+	_ "github.com/algorand/indexer/importers/all"
+	_ "github.com/algorand/indexer/processors/all"
+)
+
+var (
+	logger          *log.Logger
+	defaultLogLevel string
+)
+
+// init() function for main package
+func init() {
+
+	// Setup logger
+	logger = log.New()
+
+	formatter := conduit.PluginLogFormatter{
+		Formatter: &log.JSONFormatter{
+			DisableHTMLEscape: true,
+		},
+		Type: "Conduit",
+		Name: "main",
+	}
+
+	logger.SetFormatter(&formatter)
+	logger.SetOutput(os.Stdout)
+	defaultLogLevel = "info"
+}
+
+// conduitCmd creates the main cobra command, initializes flags, and viper aliases
+func conduitCmd() *cobra.Command {
+	cfg := &conduit.Config{}
+	conduitCmd := &cobra.Command{
+		Use:   "conduit",
+		Short: "run the conduit framework",
+		Long:  "run the conduit framework",
+		Args:  cobra.NoArgs,
+		RunE: func(cmd *cobra.Command, args []string) error {
+
+			// Tie cobra variables into local go-lang variables
+			config.BindFlagSet(cfg.Flags)
+			logger.Info(cfg)
+
+			if err := cfg.Valid(); err != nil {
+				return err
+			}
+
+			pCfg, err := conduit.MakePipelineConfig(logger, cfg)
+
+			if err != nil {
+				return err
+			}
+
+			logLevel, _ := log.ParseLevel(pCfg.PipelineLogLevel)
```

Comment:
I see where you're going with the `Valid()` pattern and ignoring the error here, but it looks really out of place to ignore errors. I'm not sure we should have the extra indirection instead of verifying everything at the top.